optimum/exporters/openvino/convert.py

```diff
@@ -778,3 +788,84 @@ def _add_version_info_to_model(model: Model, library_name: Optional[str] = None)
         pass
 
     return model
+
+
+def _get_multi_modal_submodels_and_export_configs(
+    model: Union["PreTrainedModel", "TFPreTrainedModel"],
+    task: str,
+    library_name: str,
+    int_dtype: str,
+    float_dtype: str,
+    preprocessors: Optional[List[Any]] = None,
+    model_kwargs: Optional[Dict] = None,
+    stateful: bool = True,
+):
+    models_for_export = {}
+    stateful_parts = []
+
+    model_type = model.config.model_type.replace("_", "-")
+
+    if model_type == "internvl-chat" and preprocessors is not None:
+        model.config.img_context_token_id = preprocessors[0].convert_tokens_to_ids("<IMG_CONTEXT>")
+
+    if hasattr(model, "image_newline"):
+        model.config.image_newline = model.image_newline.tolist()
+    main_config_cls = TasksManager.get_exporter_config_constructor(
+        model=model, task=task, exporter="openvino", library_name=library_name
+    )
+    main_config = main_config_cls(
+        model.config, int_dtype=int_dtype, float_dtype=float_dtype, preprocessors=preprocessors
+    )
+    for behavior in main_config.SUPPORTED_BEHAVIORS:
+        model_id = f"{behavior}_model"
+        model_part_config = main_config.with_behavior(behavior)
+        model_part = main_config.get_model_for_behaviour(model, behavior)
+        models_for_export[model_id] = (model_part, model_part_config)
+        stateful_parts.append(stateful if getattr(model_part_config, "use_past", False) else False)
+    return main_config, models_for_export, stateful_parts
```

Comment:
`stateful_parts` can be inferred from `models_for_export` so can't we get this value from there instead of returning `stateful_parts` here and in `_get_submodels_and_export_configs` to pass it to `export_models` afterwards ?

Reply:
I think it maybe nontrivial and more complex if we move that on other level, where some info about model type can be already lost. e.g. seq2seq models also split on parts and have decoder with past that currently should remain stateless until ov runtime will be ready for this IR format https://github.com/huggingface/optimum-intel/pull/779 (yes, we still working on that and I belive it will happens soon). also with exploring more models, it maybe changed too, e.g. there is blip2-t5-xl model that uses encoder_decoder architecture for language model with the same working approach that also will require some changes in this logic in future

Reply:
done

Reply:
Makes sense, let's keep it that way then!

> https://github.com/huggingface/optimum-intel/pull/779 (yes, we still working on that and I belive it will happens soon)

that's very cool !